include/grpc/grpc_security.h

```diff
@@ -1117,8 +1117,23 @@ grpc_authorization_policy_provider_static_data_create(
 GRPCAPI void grpc_authorization_policy_provider_release(
     grpc_authorization_policy_provider* provider);
 
+//struct grpc_status_or_bool{
+//  grpc_status_code status;
+//  bool boolean;
+//};
+
+/**
+ * Checks if the private key matches the certificate's public key.
+ * Returns a not-OK status on failure, or a bool indicating
+ * whether the key/cert pair matches.
+ */
+GRPCAPI grpc_status_code* grpc_tls_certificate_key_match(
```

Comment:
Currently, you are only looking at the status for PrivateKeyCertMatch. We also want to check if the match failed or was successful with status OK.
Also here you are only returning the status code, but we would need more information, so we should propagate the error message too.

I would recommend updating this API to
```
GRPCAPI bool grpc_tls_certificate_key_match(const char* private_key, const char* cert_chain, grpc_status_code* code, const char** error_details);
```
Please update the definition too. And suggest moving this API to line 850.